### ISSUE TITLE
Sudden math jargon

### ISSUE THREAD

**Issue by nalply** [abuseofnotation/category-theory-illustrated#65]
In the section "Defining the singleton set using functions" suddenly there's math jargon: "if 1 is the singleton set, then we have \<formula with exclamation operator>". I am not sure whether the exclamation operator means negation, and to be sincere, I found that jarring.

Your book is excellent and please excuse me when I want to help you making the book even better, and this is my suggestion: Try to express it in natural language and what this means. Something like this:

"We know that we can represent 1 as a singleton set.  So we can express that unique property of singleton sets in the example of 1 as: For all things X one not-X  (?) exists that maps to 1, or more succinctly: \<formula>".

And the negation operator is usually ¬, and if you use something else, you probably need something like a terminology chapter where you define operators and terminology concisely. At least I tried to look for a glossary but didn't find one.

I am not sure whether I understood you correctly, so please excuse my mistakes. And also excuse me if I just missed something and all of this is a stupid mistake of mine.

**Comment by drshade** [comment]
I hit this same wall and didn't understand the "!" either. However I think its just used as "exists" in this context as far as I could tell. I.e. "For all sets there exists an arrow from that set to the the singleton set.". Its as simple as that. Agree that adding a natural language description but also removing the "!" in this case is a good idea.

**Comment by nalply** [comment]
I just reread that section considering your input and now everything fell into place. I think the exclamation mark means uniqueness, this means, it not only exists, it also exists exactly once.

"In particular, there is *exactly one* function from any other set to the singleton set."

Now I feel like I  gained nirwana, haha.

**Comment by drshade** [comment]
Ah interesting! But wait isn't any function from the source set to the singleton set exactly the same function anyway? Eg who cares how many there are - they MUST be the same. Basically there IS only one way to pick an element of the singleton set. So im not sure about the “unique” thing…

**Comment by nalply** [comment]
I researched a little bit and found this gem:

`∃!X(P(X))` is a shorthand for `∃X(P(X)∧∀Y(P(Y)⟹Y=X))` 

In natural language: there exist X where predicate P holds for X and additionally for any other Y where this predicate also holds it follows that X and Y must be same.

**Comment by drshade** [comment]
Certainly plausible - and related to the topic at hand (albeit a bit too complex for these early chapters of the book?)

However your initial concern of this being very jarry maths at this point in the text is sound - hopefully @abuseofnotation can weigh in and polish off the sharp edges here.  

(ps. isn't the authors handle perfectly ironic!)

**Comment by abuseofnotation** [comment]
Thanks, I will look into it. 

Usually, the formulas just repeat what is stated in the paragraph with natural language, in this case it's "there is exactly one function from any other set to the singleton set"

Otherwise that is standard logic notation 
- ∃ - exist
- ∃! - exist unique

**Comment by abuseofnotation** [comment]
> Ah interesting! But wait isn't any function from the source set to the singleton set exactly the same function anyway? Eg who cares how many there are - they MUST be the same. Basically there IS only one way to pick an element of the singleton set. So im not sure about the “unique” thing…

Yes, but the idea here is to define the singleton set using functions. In CT the formula is the starting point, not the inside of the object...

**Comment by abuseofnotation** [comment]
How's that?
![image](https://github.com/user-attachments/assets/a0456f61-4548-43c6-b4ac-834264e515ba)
